### PR TITLE
DEVDOCS-4575: [update] Catalog V3 API, update request response schemas

### PR DESCRIPTION
**Pull request by bc-andreadao**
# [DEVDOCS-4575]

- Removed `id` from POST request bodies. Sometimes, doing this also removed `id` from responses (that need `id`), so I had to add `id` in other places of the spec. 

- Metafield endpoints:
   1.  Removed unneeded fields from POST and PUT request bodies
   2.  Reordered fields for response examples 
   3.  Fixed `resource_type` for response examples (should correspond to the endpoint) 

[DEVDOCS-4575]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4575?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reference/catalog.v3.yml [hunk 1]
```diff
@@ -24180,13 +23244,29 @@ components:
...
+                - product
+                - variant
+              x-required:
+                - post
+            resource_id:
+              maximum: 10000000000
+              minimum: 0
+              type: integer
+              description: |
+                The ID for the resource with which the metafield is associated.
```
Comment (bc-tgomez) [inline reference/catalog.v3.yml:23266]:
```suggestion
                The ID of the resource with which the metafield is associated.
```

File: reference/catalog.v3.yml [hunk 2]
```diff
@@ -22474,44 +21576,10 @@ components:
       x-internal: false
     metafield_Base:
       title: metafield_Base
-      required:
-        - key
-        - namespace
-        - permission_set
-        - value
       type: object
+      description: 'Metafield for products, categories, variants, and brands. The max number of metafields allowed on each product, category, variant, or brand is fifty. For more information, see [Platform Limits](https://support.bigcommerce.com/s/article/Platform-Limits) in the Help Center.'
```
Comment (bc-tgomez) [inline reference/catalog.v3.yml:21580]:
```suggestion
      description: Metafield for products, categories, variants, and brands; the max number of metafields allowed on each is 50. For more information, see [Platform Limits](https://support.bigcommerce.com/s/article/Platform-Limits) in the Help Center.
```

File: reference/catalog.v3.yml [hunk 3]
```diff
@@ -16280,13 +15690,9 @@ paths:
           application/json:
             schema:
               title: Brand
-              required:
-                - name
               type: object
+              description: Common Brand properties.
```
Comment (bc-tgomez) [inline reference/catalog.v3.yml:15694]:
```suggestion
              description: Common brand properties.
```

**Review by bc-tgomez — APPROVED** [review]
It all looks good! I made some suggestions that you can take or leave. 😃

**Review by bc-traciporter — COMMENTED** [review]
Looks good Andrea. I was not aware of your PR when I address one of these issues with the id in the request for [PR 1112](https://github.com/bigcommerce/api-specs/pull/1112). Do you think I should cancel mine? We went about resolving this issue a different way and I don't want to cause conflicts.

**Comment by bc-andreadao** [comment]
> Looks good Andrea. I was not aware of your PR when I address one of these issues with the id in the request for [PR 1112](https://github.com/bigcommerce/api-specs/pull/1112). Do you think I should cancel mine? We went about resolving this issue a different way and I don't want to cause conflicts.

@bc-traciporter Yes, this PR should solve PR 1112, thanks for mentioning that in your ticket earlier

File: reference/catalog.v3.yml (hunk 3 above)
Comment (bc-andreadao) [reply reference/catalog.v3.yml:15694]: Fixed it to your suggestion

File: reference/catalog.v3.yml (hunk 2 above)
Comment (bc-andreadao) [reply reference/catalog.v3.yml:21580]: Fixed it to your suggestion

File: reference/catalog.v3.yml (hunk 1 above)
Comment (bc-andreadao) [reply reference/catalog.v3.yml:23266]: Fixed it to your suggestion

File: reference/catalog.v3.yml [hunk 4]
```diff
@@ -4249,123 +4246,24 @@ paths:
...
                   id: 4
                   key: location_id
+                  value: 'Shelf 3, Bin 5'
                   namespace: App Namespace
                   permission_set: app_only
+                  resource_type: variant
                   resource_id: 137
-                  resource_type: product
-                  value: 'Shelf 3, Bin 5'
+                  description: Where products are located
```
Comment (slsriehl) [inline reference/catalog.v3.yml:4264]:
```suggestion
                  description: Where products are located.
```

File: reference/catalog.v3.yml [hunk 5]
```diff
@@ -11663,7 +11542,15 @@ paths:
                   data:
                     type: array
                     items:
-                      $ref: '#/components/schemas/bulkPricingRule_Full'
+                      allOf:
+                        - type: object
+                          properties:
+                            id:
+                              type: integer
+                              description: Unique ID of the *Bulk Pricing Rule*. Read-Only.
```
Comment (slsriehl) [inline reference/catalog.v3.yml:11550]:
```suggestion
                              description: Unique ID of the *Bulk Pricing Rule*. Read-Only.
                              readOnly: true
```

File: reference/catalog.v3.yml [hunk 6]
```diff
@@ -11777,11 +11664,18 @@ paths:
                 type: object
                 properties:
                   data:
-                    $ref: '#/components/schemas/bulkPricingRule_Full'
+                    allOf:
+                      - type: object
+                        properties:
+                          id:
+                            type: integer
+                            description: Unique ID of the *Bulk Pricing Rule*. Read-Only.
```
Comment (slsriehl) [inline reference/catalog.v3.yml:11672]:
```suggestion
                            description: Unique ID of the *Bulk Pricing Rule*. Read-Only.
                            readOnly: true
```

File: reference/catalog.v3.yml [hunk 7]
```diff
@@ -11928,7 +11822,15 @@ paths:
...
```
Comment (slsriehl) [inline reference/catalog.v3.yml:11830]:
```suggestion
                            description: Unique ID of the *Bulk Pricing Rule*. Read-Only.
                            readOnly: true
```

File: reference/catalog.v3.yml [hunk 8]
```diff
@@ -12012,7 +11914,15 @@ paths:
         content:
           application/json:
             schema:
-              $ref: '#/components/schemas/bulkPricingRule_Full'
+              allOf:
+                - type: object
+                  properties:
+                    id:
+                      type: integer
+                      description: Unique ID of the *Bulk Pricing Rule*. Read-Only.
```
Comment (slsriehl) [inline reference/catalog.v3.yml:11922]:
```suggestion
                      description: Unique ID of the *Bulk Pricing Rule*. Read-Only.
                      readOnly: true
```

File: reference/catalog.v3.yml [hunk 9]
```diff
@@ -23215,6 +22277,12 @@ components:
     product_Full:
       title: product_Full
       allOf:
+        - type: object
+          properties:
+            id:
+              minimum: 1
+              type: integer
+              description: ID of the product. Read Only.
```
Comment (slsriehl) [inline reference/catalog.v3.yml:22285]:
```suggestion
              description: ID of the product. Read-only.
              readOnly: true
```

File: reference/catalog.v3.yml [hunk 10]
```diff
@@ -24155,7 +23212,15 @@ components:
         bulk_pricing_rules:
           type: array
           items:
-            $ref: '#/components/schemas/bulkPricingRule_Full'
+            allOf:
+              - type: object
+                properties:
+                  id:
+                    type: integer
+                    description: Unique ID of the *Bulk Pricing Rule*. Read-Only.
```
Comment (slsriehl) [inline reference/catalog.v3.yml:23220]:
```suggestion
                    description: Unique ID of the *Bulk Pricing Rule*. Read-Only.
                    readOnly: true
```

**Review by slsriehl — APPROVED** [review]
nice work.

**Pull request merged**